### PR TITLE
rocksdb: package binary tools

### PR DESCRIPTION
**Pull request by yorickvP**
###### Motivation for this change

rocksdb includes some binaries that can be useful for debugging (sst_dump, ldb, rocksdb_dump, rocksdb_undump, blob_dump).
Add a bin output and install them there.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/rocksdb/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ stdenv
+{ stdenv, lib
```
Comment (yegortimoshenko) [inline pkgs/development/libraries/rocksdb/default.nix]: Use `stdenv.lib` instead.

**Comment by yegortimoshenko** [comment]
@GrahamcOfBorg build rocksdb

File: pkgs/development/libraries/rocksdb/default.nix (hunk 1 above)
Comment (cstrahan) [reply pkgs/development/libraries/rocksdb/default.nix]: @yegortimoshenko  I always write `lib` as a formal argument in my expressions; have we made a formal decision somewhere to prefer `stdenv.lib` over a separate `lib` argument?

Comment (yegortimoshenko) [reply pkgs/development/libraries/rocksdb/default.nix]: I've seen other reviewers ask that (@joachifm, @orivej), but I haven't seen a discussion.

Comment (joachifm) [reply pkgs/development/libraries/rocksdb/default.nix]: I'd wager that `stdenv.lib` vastly  outnumbers passing `lib` separately (for package expressions). I'm unaware of any arguments beyond uniformity.

Whether it makes sense for `stdenv` to provide `lib` is debatable I suppose, but it *does* & is widely relied upon, so to me it makes more sense to just stick with convention for convention's sake.

**Pull request merged**